src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -78,6 +85,39 @@ public Response createDaaForDac(
     }
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+  @Path("{daaId}/{libraryCardId}")
+  public Response createLibraryCardDaaRelation(
+      @Context UriInfo info,
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId,
+      @PathParam("libraryCardId") Integer libraryCardId) {
+    try {
+      DataAccessAgreement daa = daaService.findById(daaId);
+      User user = userService.findUserByEmail(authUser.getEmail());
+      int userInstitutionId = user.getInstitutionId();
+      LibraryCard libraryCard = libraryCardService.findLibraryCardById(libraryCardId);
+      int libraryCardInstitutionId = libraryCard.getInstitutionId();
+      // Assert that the user has the correct institution permissions to add a DAA-LC relationship.
+      // Admins can add a DAA with any DAC, but signing officials can only create relationships for
+      // library cards associated with the same institution they are associated with.
+      if (!user.hasUserRole(UserRoles.ADMIN)) {
+        if (!(userInstitutionId == libraryCardInstitutionId)) {
+          return Response.status(Status.FORBIDDEN).build();
+        }
+      }
+      libraryCardService.addDaaToLibraryCard(libraryCardId, daaId);
+      URI uri = info.getBaseUriBuilder()
+          .replacePath("api/libraryCards/{libraryCardId}")
+          .build(libraryCardId);
+      return Response.created(uri).entity(libraryCard).build();
```

Comment:
I don't think this is the right response, we didn't create a library card, we just associated an existing one to an existing DAA. I think this could be simplified to an OK response. The location header is fine, and in fact, we should do better about that in our endpoints.